cfme/tests/candu/test_utilization_metrics.py

```diff
@@ -302,3 +308,71 @@ def test_raw_metric_host_disk(metrics_collection, appliance, provider):
         if record.disk_usage_rate_average is not None:
             assert record.disk_usage_rate_average > 0, 'Zero Host Disk IO'
             break
+
+
+def query_metric_rollup_table(appliance, provider, metric, azone_name):
+    metrics_tbl = appliance.db.client['metric_rollups']
+    ems = appliance.db.client['ext_management_systems']
+
+    with appliance.db.client.transaction:
+        provs = (
+            appliance.db.client.session.query(metrics_tbl.id)
+            .join(ems, metrics_tbl.parent_ems_id == ems.id)
+            .filter(metrics_tbl.resource_name == azone_name,
+            ems.name == provider.name)
+        )
+    return appliance.db.client.session.query(metrics_tbl).filter(
+        metrics_tbl.id.in_(provs.subquery()))
+
+
+def generic_test_azone_rollup(appliance, provider, metric, azone_name):
+    query = query_metric_rollup_table(appliance, provider, metric, azone_name)
+
+    for record in query:
+        if getattr(record, metric) is not None:
+            assert getattr(record, metric) > 0, (
+                'Zero Azone CPU usage/memory usage/Disk IO/Network IO'
+            )
+            break
+
+
+@pytest.mark.uncollectif(lambda provider:
+                         provider.one_of(RHEVMProvider, VMwareProvider))
+def test_azone_candu_cpu(metrics_collection, appliance, provider):
+    """
+    Polarion:
+        assignee: nachandr
+        caseimportance: high
+        casecomponent: CandU
+        initialEstimate: 1/12h
+    """
+    azone_name = provider.data["cap_and_util"]["capandu_azone"]
+    generic_test_azone_rollup(appliance, provider, 'cpu_usage_rate_average', azone_name)
+
+
+@pytest.mark.uncollectif(lambda provider:
+                         provider.one_of(RHEVMProvider, VMwareProvider))
+def test_azone_candu_memory(metrics_collection, appliance, provider):
+    """
+    Polarion:
+        assignee: nachandr
+        caseimportance: high
+        casecomponent: CandU
+        initialEstimate: 1/12h
+    """
+    azone_name = provider.data["cap_and_util"]["capandu_azone"]
+    generic_test_azone_rollup(appliance, provider, 'mem_usage_absolute_average', azone_name)
+
+
+@pytest.mark.uncollectif(lambda provider:
+                         provider.one_of(RHEVMProvider, VMwareProvider))
+def test_azone_candu_network(metrics_collection, appliance, provider):
+    """
+    Polarion:
+        assignee: nachandr
+        caseimportance: high
+        casecomponent: CandU
+        initialEstimate: 1/12h
+    """
+    azone_name = provider.data["cap_and_util"]["capandu_azone"]
+    generic_test_azone_rollup(appliance, provider, 'net_usage_rate_average', azone_name)
```

Comment:
@nachandr **Required** since these tests use the same providers and steps I recommend parametrizing them like

```python
@pytest.mark.parametrize("metric", ['cpu_usage_rate_average', 'mem_usage_absolute_average', 'net_usage_rate_average'], ids=['cpu', 'mem', 'network'])
```

Reply:
The three new tests require that the metrics_collection fixture be run. The first metric collection is scheduled through this  fixture. The first metric collection itself takes at least 15 minutes(sometimes longer). The three tests that are run per provider take at least  20 minutes each for completion(considering  provider add, provider refresh etc.). But, without parametrization, the three tests could bee run in under 20 minutes.So, I did not parametrize the tests.

I should have added a note about this in the PR when I submitted the PR.

Reply:
Ah okay, thanks for the explanation!